pykern/pkcollections.py

```diff
@@ -99,9 +101,77 @@ def pkdel(self, name, default=None):
             except KeyError:
                 pass
 
+    def pkmerge(self, to_merge):
+        """Add `to_merge` to `self`
+
+        `self` will have references to `to_merge` after this call.
+        If you want a pure copy, use `canonicalize` or `copy.deepcopy` first.
+
+        Types are assumed to match and are not converted, e.g. dict is
+        not converted to PKDict. Again, use `canonicalize` if you want
+        to avoid type incompatibilities.
+
+        `to_merge`'s values override `self`'s so if say, `to_merge` is ``{'x': None}``,
+        then ``self.x`` will be `None` at the end of this call even if it had a value
+        before.
+
+        Lists from to_merge are prepended on this same principle, that
+        is, to_merge "overrides" self, and prepending is defined as
+        overriding. Lists must contain unique elements and duplicates will
+        cause an error.
+
+        This function recurses only on PKDicts.
+
+        Args:
+            to_merge (dict): elements will be copied into `self`
+
+        """
+
+        def _type_err(key, base, merge):
+            return AssertionError(
+                f"key={key} type mismatch between (self) base={base} and to_merge={merge}"
+            )
+
+        for k in list(to_merge.keys()):
+            t = to_merge[k]
+            s = self.get(k)
+            if isinstance(s, dict) or isinstance(t, dict):
+                if s is None:
+                    pass
+                elif t is None:
```

Comment:
I think this can be collapsed into the first `if` like you do for the list check below

Reply:
Yes, see https://github.com/radiasoft/pykern/pull/224#discussion_r910031818